.github/workflows/release.yml

```diff
@@ -0,0 +1,72 @@
+name: CI_CD_ON_TAG
+
+on:
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+      - 'v*.*.*' # Push events to matching v*.*.*, i.e. v0.0.1, v2.1.3
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Checkout single line diagram sources
```

Comment:
@ChamseddineBhd we can see here that you have a problem with your release design as you should not be obliged to buill master of dependent librairies but use a release instead.